### ISSUE TITLE
docker system df seems slow

### ISSUE THREAD

**Issue by jamesongithub** [moby/moby#31719]
**Description**
`docker system df` command seems slow`. I just launched 20 or 30 containers on a fresh host and ran system df. It took 2 minutes to return results.

**Steps to reproduce the issue:**
1. Run 20 containers.
2. Run `docker system df`. Time it

**Describe the results you received:**
```
time docker system df
TYPE                TOTAL               ACTIVE              SIZE                RECLAIMABLE
Images              19                  13                  10.29 GB            5.262 GB (51%)
Containers          26                  19                  14.21 MB            28.17 kB (0%)
Local Volumes       1                   1                   0 B                 0 B

real	2m57.581s
user	0m0.015s
sys	0m0.000s
```

**Describe the results you expected:**
Less than 2 minutes. Ideally as fast as `docker info`

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
docker --version
Docker version 1.13.1, build 092cba3
```

**Output of `docker info`:**

```
Containers: 24
 Running: 23
 Paused: 0
 Stopped: 1
Images: 19
Server Version: 1.13.1
Storage Driver: devicemapper
 Pool Name: thinpoolname
 Pool Blocksize: 65.54 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: ext4
 Data file: 
 Metadata file: 
 Data Space Used: 16.57 GB
 Data Space Total: 192.4 GB
 Data Space Available: 175.8 GB
 Metadata Space Used: 17.42 MB
 Metadata Space Total: 2.751 GB
 Metadata Space Available: 2.734 GB
 Thin Pool Minimum Free Space: 19.24 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Library Version: 1.02.77 (2012-10-15)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge host macvlan null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: aa8187dbd3b7ad67d8e5e3a15115d3eef43a7ed1
runc version: 9df8b306d01f59d3a8029be411de015b7304dd8f
init version: 949e6fa
Security Options:
 apparmor
Kernel Version: 3.13.0-96-generic
Operating System: Ubuntu 14.04.5 LTS
OSType: linux
Architecture: x86_64
CPUs: 8
Total Memory: 59.97 GiB
Name: hostname
ID: REDACT
Docker Root Dir: REDACT
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 181
 Goroutines: 188
 System Time: 2017-03-10T02:08:06.201928278Z
 EventsListeners: 1
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Comment by floushee** [comment]
I have the same problem:

time docker system df
TYPE                TOTAL               ACTIVE              SIZE                RECLAIMABLE
Images              14                  8                   4.295GB             3.461GB (80%)
Containers          8                   8                   196.9MB             0B (0%)
Local Volumes       0                   0                   0B                  0B

real	5m27.818s
user	0m0.007s
sys	0m0.018s


docker version
Client:
 Version:      17.04.0-ce
 API version:  1.28
 Go version:   go1.7.5
 Git commit:   4845c56
 Built:        Mon Apr  3 18:01:50 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.04.0-ce
 API version:  1.28 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   4845c56
 Built:        Mon Apr  3 18:01:50 2017
 OS/Arch:      linux/amd64
 Experimental: false

docker info
Containers: 8
 Running: 8
 Paused: 0
 Stopped: 0
Images: 129
Server Version: 17.04.0-ce
Storage Driver: devicemapper
 Pool Name: docker-253:0-1146056-pool
 Pool Blocksize: 65.54kB
 Base Device Size: 10.74GB
 Backing Filesystem: xfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 6.915GB
 Data Space Total: 107.4GB
 Data Space Available: 27.49GB
 Metadata Space Used: 11.06MB
 Metadata Space Total: 2.147GB
 Metadata Space Available: 2.136GB
 Thin Pool Minimum Free Space: 10.74GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.135-RHEL7 (2016-11-16)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge host macvlan null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: 
containerd version: 422e31ce907fd9c3833a38d7b8fdd023e5a76e73
runc version: 9c2d8d184e5da67c95d601382adf14862e4f2228
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 3.10.0-514.10.2.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 7.64GiB
Name: hostname
ID: GTJS:EWDM:DKJX:5YHI:JQ76:KQEA:XKTL:7IUP:YTO7:BZMR:BTBR:LPSJ
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

**Comment by thaJeztah** [comment]
This is a known limitation of the `system df` command, also see https://github.com/moby/moby/pull/31973, and https://github.com/moby/moby/issues/31951

Although this issue predates https://github.com/moby/moby/issues/31951, I'm gonna close this one because the other issue has a more active discussion. Please subscribe to that issue to hear about possible improvements in this area.

I'll lock the conversation on this issue, to prevent the discussion from diverging
